src/light_data/light_store.rs

```diff
@@ -137,30 +141,49 @@ impl<F: LurkField> LightStore<F> {
     fn to_scalar_store(&self) -> anyhow::Result<ScalarStore<F>> {
         let mut store = ScalarStore::default();
         for (ptr, le) in self.scalar_map.iter() {
-            let se = match le {
-                None => None,
-                Some(LightExpr::Cons(x, y)) => Some(ScalarExpression::Cons(*x, *y)),
-                Some(LightExpr::Comm(f, x)) => Some(ScalarExpression::Comm(*f, *x)),
-                Some(LightExpr::Num(f)) => Some(ScalarExpression::Num(*f)),
-                Some(LightExpr::Char(f)) => {
+            let tag: ExprTag = ExprTag::from_field(ptr.tag()).ok_or(anyhow!("expr_tag"))?;
```

Comment:
Consider using `ok_or_else`, as the anyhow macro isn't free. A slightly longer error messages, such as "invalid expr_tag" would also help.